### DISCUSSION TITLE
Question (and potential bug) about the JS SDK

### DISCUSSION THREAD

**Discussion by JunoNgx** [pocketbase/pocketbase#4616] [Q&A] [answered]
I'm aware this is not the repository for the JS SDK, but there are a lot more activities here. Please let me know if I should repost this elsewhere.

Context: I'm implementing an auto-save feature in a modal, where pocketbase client will make a call to update the record with the current text fields, just before closing. This is called in the clean up function of `useEffect` of a React app.

```
    useEffect(() => {
        return () => {
            saveOnClose();
        }
    }, []);

    const saveOnClose = async () => {
        if (hasChangedRef.current && !hasSavedRef.current) {
            await updateItemTitleAndContent({
                itemId: item.id,
                title: titleValRef.current,
                content: contentValRef.current,
            });
        }
    }
```
![Screenshot from 2024-03-23 19-08-52](https://github.com/pocketbase/pocketbase/assets/5671813/e227c937-3d6a-4fa6-9258-30e82b592299)

Question 1: What is the `OPTIONS` request that comes with every single `PATCH` request?

Question 2: My `PATCH` request is consistently auto-cancelled in this particular case. There is no second request or duplicated one. I tried running without `React.StrictMode`, but it was the same.

The only way to get this to run as I intended is to aggressively disable auto-cancellation, by either `pbClient.autoCancellation(false)` or setting `{requestKey: null}`.

![Screenshot from 2024-03-23 19-17-57](https://github.com/pocketbase/pocketbase/assets/5671813/ad498a98-1abf-4c00-9d0e-e7c5dd101dfa)

I'm not sure if this is how it's supposed to work. Is it a bug, or might I have been doing something wrong?

**Comment by ganigeorgiev** [comment]
The `OPTIONS` request is send automatically by the browser to check whether the request is CORS allowed.

The `PATCH` request fails because probably the `OPTIONS` one didn't pass or some other reasons, but it doesn't seem to be because of autoCancellation or at least the above screenshot doesn't show the duplicated call (setting `autoCancellation(false)` disables autoCancellation entirely for the current SDK instance).

Why the request doesn't work in your case I'm not sure. Try to wrap your request in a `try/catch` block and inspect the catched err and more specifically `err.originalError` (for all error fields see [Error handling](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#error-handling)).

**Reply by ganigeorgiev** [reply]
Additionally for more details about how autocancellation works, you can refer to https://github.com/pocketbase/js-sdk?tab=readme-ov-file#auto-cancellation.

**Reply by JunoNgx** [reply, marked as answer]
This is the content of `err.originalError`.
![Screenshot from 2024-03-23 20-25-43](https://github.com/pocketbase/pocketbase/assets/5671813/bff6db45-9ddf-4a4f-b068-026b83a6a88a)
This did help me track down the root cause. This was because one of my very reckless `.cancelAllRequests()`, which was called right after the `PATCH` call.

Thank you very much for your help.
